frame/evm/src/runner/stack.rs

```diff
@@ -629,6 +643,15 @@ where
 	}
 
 	fn set_storage(&mut self, address: H160, index: H256, value: H256) {
```

Comment:
> If self.original_storage does not contain the key,
That's what `if let Vacant(e) = self.original_storage.entry(...)` does, which was recommanded by clippy instead of `contains` then `insert`.
> and the to-set-value is the same as the current value.
Yeah, can add this check to not insert for same value as original.